### ISSUE TITLE
Option to disable storing pages in temp folder

### ISSUE THREAD

**Issue by lambdareader** [Difegue/LANraragi#979]
I couldn't find any information in the docs to disable it.
Setting temp folder size to 0 doesn't work. 
I don't have it on SSD so it just creates a copy on my hard drive for no reason at all.

Most manga/comic readers (komga, kavita, yacreader) just read/unzip single pages of a cbz in memory through random access. It is much faster than unpacking it, writing it and then reading.

Could this be implemented please? It just creates a lot of extra hardware strain and the gains are negligible for a single user. Even with SSD it wouldn't be worth it. Hard drives take a fraction of a second to read a few MB for a page if you have them on a NAS spinning the entire time.  
tbh I'm confused why caching is standard. It would only make sense if there are multiple users or you are on desktop and the drive platter are spinning down when not used. But aren't most using this on a NAS?
Sure it would use a bit of memory to unzip pages in memory but isn't this a case where it should be used?

**Comment by Difegue** [comment]
👋 The full extract/caching was designed to avoid performing the unarchiving job which (used) to take a bunch of time.. This is a pretty old part of the design, the server used to rely on `unar` instead of libarchive and that couldn't do random access.  

The server does support random access when you hit individual page endpoints, but it's true that it always copies to the filesystem first. I'm not sure how much work it'd take to make it all happen in-memory.  
As a workaround, I think you could just point the temp folder to a RAM disk?

**Comment by lambdareader** [comment]
Thanks for the fast reply. 
Wow, stupid of me to not think of simply using a ramdisk. I'll be doing that, though always having to load all pages at once is a bit wasteful. 
It would be pretty neat to eventually rewrite it.

As another thought. I'm running lanraragi on a very low powered device. Is there a option to just generate page thumbnails if i open the archive overview and not when i open the archive (or disable them entirely)? It causes the server to lag because all cores are at 100% if there are triple digit pages. I don't really use the overview very much, so every time i want to read a big archive it takes 30 seconds before i can turn to the next page.

**Comment by Difegue** [comment]
No specific option, but I believe the changes from #885 will help as they restrain page thumbnail jobs to run sequentially on one worker only.

**Comment by Asinin3** [comment]
Yeah, this would be great if we can just disable the cache saving to disk. A lot of people use LRR locally, and I also have all of my archives uncompressed, using the store level of compression. Which means they should be fast to open, and the caching of galleries is fairly redundant.

**Comment by siliconfeces** [comment]
FWIW, I'm doing some related experiments but can't promise it will pan out to something useful. LRR is somewhat based on letting pages always touch disk, so reworking that without breaking a bazillion things is a bit hairy.

**Comment by Asinin3** [comment]
Well making a RAM Disk on windows requires a thirdparty program, would rather avoid.
